### PR TITLE
ENT-6876: Optimised vault query to not query for total state count if the first page isn't full

### PR DESCRIPTION
**Pull request by shamsasari**

### CODE REVIEW

**Comment by shamsasari** [comment]
Once approved, will then kick off the 4.8 OS to ENT merge and the forward merge up for both repos.

File: node/src/main/kotlin/net/corda/node/services/vault/NodeVaultService.kt [hunk 1]
```diff
@@ -732,6 +729,13 @@ class NodeVaultService(
                 ArrayList()
         )
 
+        val totalStatesAvailable = when {
+            paging.isDefault -> -1L
```
Comment (rick-r3) [inline node/src/main/kotlin/net/corda/node/services/vault/NodeVaultService.kt:733]: Is this really safe?  i.e. page = 1 and page_size = 200.   I'd be inclined to just leave this as is and just have the optimisation on the line below (first page and less results than page size).

Comment (shamsasari) [reply node/src/main/kotlin/net/corda/node/services/vault/NodeVaultService.kt:733]: I'm not sure I follow. I've not changed the behaviour for the `isDefault` case.

Comment (rick-r3) [reply node/src/main/kotlin/net/corda/node/services/vault/NodeVaultService.kt:733]: Oh yeah.

**Review by rick-r3 — APPROVED** [review]

**Pull request merged**